featurebyte/app.py

```diff
@@ -55,46 +55,70 @@
 logger = get_logger(__name__)
 
 
-def _get_api_deps() -> Callable[[Request, PydanticObjectId], None]:
+def _dep_injection_func(
+    request: Request, active_catalog_id: Optional[PydanticObjectId] = None
+) -> None:
+    """
+    Inject dependencies into the requests
+
+    Parameters
+    ----------
+    request: Request
+        Request object to be updated
+    active_catalog_id: Optional[PydanticObjectId]
+        Catalog ID to be used for the request
+    """
+    request.state.persistent = get_persistent()
+    request.state.user = User()
+    request.state.get_credential = MongoBackedCredentialProvider(
+        persistent=request.state.persistent
+    ).get_credential
+    request.state.app_container = LazyAppContainer(
+        user=request.state.user,
+        persistent=request.state.persistent,
+        temp_storage=get_temp_storage(),
+        celery=get_celery(),
+        storage=get_storage(),
+        catalog_id=active_catalog_id,
+        app_container_config=app_container_config,
+    )
+
+
+def _get_api_deps() -> Callable[[Request], Coroutine[Any, Any, None]]:
     """
     Get API dependency injection function
 
     Returns
     -------
-    Callable[Request, PydanticObjectId]
+    Callable[Request]
         Dependency injection function
     """
 
-    def _dep_injection_func(
-        request: Request, active_catalog_id: PydanticObjectId = Header(None)
+    async def _wrapper(
+        request: Request,
     ) -> None:
-        """
-        Inject dependencies into the requests
+        _dep_injection_func(request)
 
-        Parameters
-        ----------
-        request: Request
-            Request object to be updated
-        active_catalog_id: PydanticObjectId
-            Catalog ID from header
-        """
-        active_catalog_id = active_catalog_id or PydanticObjectId(DEFAULT_CATALOG_ID)
-        request.state.persistent = get_persistent()
-        request.state.user = User()
-        request.state.get_credential = MongoBackedCredentialProvider(
-            persistent=request.state.persistent
-        ).get_credential
-        request.state.app_container = LazyAppContainer(
-            user=request.state.user,
-            persistent=request.state.persistent,
-            temp_storage=get_temp_storage(),
-            celery=get_celery(),
-            storage=get_storage(),
-            catalog_id=active_catalog_id,
-            app_container_config=app_container_config,
-        )
+    return _wrapper
+
+
+def _get_api_deps_with_catalog() -> Callable[[Request], Coroutine[Any, Any, None]]:
```

Comment:
Split dependency injection function into catalog-specific and non catalog-specific versions.
Active catalog header field will only appear for catalog-specific routes in OpenAPI specs instead of all routes.